overlord/devicestate/handlers.go

```diff
@@ -72,6 +74,17 @@ func (m *DeviceManager) doMarkPreseeded(t *state.Task, _ *tomb.Tomb) error {
 				if _, err := exec.Command("umount", "-d", "-l", info.MountDir()).CombinedOutput(); err != nil {
 					return err
 				}
+				// Remove early mount for the kernel snap
+				// TODO we need something like EarlyKernelPlaceInfo
+				// to avoid repeating the kernel-snaps path.
+				if tp, _ := snapSt.Type(); tp == snap.TypeKernel {
+					earlyMntPt := filepath.Join("/run/mnt/kernel-snaps",
```

Comment:
some public helper either in boot or in kernel would be great to add for this in a followup

Reply:
maybe kernel